File: Backend/clinic/src/main/java/com/example/demo/entity/Specialty.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "specialties")
@Data
public class Specialty {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "clinic_id")
    private Long clinicId;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "icon_url")
    private String iconUrl;

    @Column(name = "head_doctor_id")
    private Long headDoctorId; // Định danh Trưởng khoa
}

File: BenhNhan/myapp/lib/models/specialty.dart
class SpecialtyModel {
  final int? id;
  final String name;
  final String? description;
  final String? iconUrl;
  final int? headDoctorId;

  SpecialtyModel({
    this.id,
    required this.name,
    this.description,
    this.iconUrl,
    this.headDoctorId,
  });

  factory SpecialtyModel.fromJson(Map<String, dynamic> json) {
    return SpecialtyModel(
      id: json['id'],
      name: json['name'] ?? '',
      description: json['description'],
      iconUrl: json['iconUrl'],
      headDoctorId: json['headDoctorId'],
    );
  }
}